File: common/src/main/java/chat/common/nimbl3/com/presenter/MessagesPresenter.java
package chat.common.nimbl3.com.presenter;

import java.util.List;

import chat.common.nimbl3.com.model.Message;
import chat.common.nimbl3.com.service.Api;
import chat.common.nimbl3.com.service.GithubApi;
import chat.common.nimbl3.com.view.IMessageView;
import rx.Scheduler;
import rx.Subscriber;

public class MessagesPresenter {

    IMessageView mView;

    public MessagesPresenter(IMessageView view) {
        this.mView = view;
    }

    public void getMessages(Scheduler postThread, Scheduler subscribeThread) {
        GithubApi gitHubApi = Api.createService(GithubApi.class);
        gitHubApi.getRepos()
                .subscribeOn(subscribeThread)
                .observeOn(postThread)
                .subscribe(new Subscriber<List<Message>>() {
                    @Override
                    public void onCompleted() {
                    }

                    @Override
                    public void onError(Throwable e) {
                    }

                    @Override
                    public void onNext(List<Message> repositories) {
                        mView.showMessages(repositories);
                    }
                });
    }
}

File: common/src/main/java/chat/common/nimbl3/com/presenter/MessagesPresenter.kt
package chat.common.nimbl3.com.presenter

import chat.common.nimbl3.com.model.Message
import chat.common.nimbl3.com.service.Api
import chat.common.nimbl3.com.iview.IMessageView
import rx.Scheduler
import rx.Subscriber

class MessagesPresenter(internal var mView: IMessageView) {

    fun getMessages(postThread: Scheduler, subscribeThread: Scheduler) {
        Api.service.repos
                .subscribeOn(subscribeThread)
                .observeOn(postThread)
                .subscribe(object : Subscriber<List<Message>>() {
                    override fun onCompleted() {}

                    override fun onError(e: Throwable) {}

                    override fun onNext(repositories: List<Message>) {
                        mView.showMessages(repositories)
                    }
                })
    }
}
